core/Rename.cpp

```diff
@@ -266,30 +267,62 @@ namespace olympia
                 renaming_inst->setStatus(Inst::Status::RENAMED);
                 ILOG("sending inst to dispatch: " << renaming_inst);
 
-                // TODO: Register renaming for sources
                 const auto & srcs = renaming_inst->getSourceOpInfoList();
+                const auto & dests = renaming_inst->getDestOpInfoList();
                 for(const auto & src : srcs)
                 {
-                    const auto rf  = olympia::coreutils::determineRegisterFile(src);
-                    const auto num = src.field_value;
-                    auto & bitmask = renaming_inst->getSrcRegisterBitMask(rf);
-                    const uint32_t prf = map_table_[rf][num];
-                    reference_counter_[rf][prf]++;
-                    renaming_inst->getRenameData().setSource({prf, rf});
-                    bitmask.set(prf);
-                    ILOG("\tsetup source register bit mask "
-                         << sparta::printBitSet(bitmask)
-                         << " for '" << rf << "' scoreboard");
-                }
+                    if(renaming_inst->isLoadStoreInst()){
+                        if(src.field_id == mavis::InstMetaData::OperandFieldID::RS2){
```

Comment:
Pls add comments on what this code is doing.

Reply:
Comments added in [5212a4c](https://github.com/riscv-software-src/riscv-perf-model/pull/70/commits/5212a4c3274e758abca7bebf8ebc45a4ac210ca9)